nexus/src/main/kotlin/com/vanniktech/maven/publish/nexus/Nexus.kt

```diff
@@ -48,12 +48,24 @@ class Nexus(
       throw IllegalArgumentException("No staging profiles found in account $username. Make sure you called \"./gradlew publish\".")
     }
 
-    val candidateProfiles = allProfiles.filter { group == it.name || group.startsWith(it.name) }
+    if (allProfiles.size == 1) {
+      return allProfiles[0]
+    }
+
+    var candidateProfiles = allProfiles.filter { group == it.name }
+
+    if (candidateProfiles.isEmpty()) {
+      candidateProfiles = allProfiles.filter { group.startsWith(it.name) }
+    }
+
+    if (candidateProfiles.isEmpty()) {
+      candidateProfiles = allProfiles.filter { group.commonPrefixWith(it.name).isNotEmpty() }
+    }
```

Comment:
This can be done in one-liner as...

```kotlin
val candidateProfiles = allProfiles.filter { group == it.name }
    .ifEmpty { allProfiles.filter { group.startsWith(it.name) } }
    .ifEmpty { allProfiles.filter { group.commonPrefixWith(it.name).isNotEmpty() } }
```

Reply:
👍 updated it [e2a725f](https://github.com/vanniktech/gradle-maven-publish-plugin/pull/498/commits/e2a725f81737b9506d9f5e29fda774f4060fe21e)